### PR TITLE
Update and rename Samsung.xml to Samsung.com.xml

### PR DESCRIPTION
**Pull request by renyouguo**

### CODE REVIEW

File: src/chrome/content/rules/AllShare_Play.xml [hunk 1]
```diff
@@ -1,33 +1,26 @@
...
-
+	<target host="allshareplay.com" />
+	<target host="www.allshareplay.com" />
+	<target host="fwk.allshareplay.com" />
+	<target host="m.allshareplay.com" />
+		<!--	403 in http://m.allshareplay.com/	-->
+		<exclusion pattern="^http://m\.allshareplay\.com/$" />
+			<test url="http://m.allshareplay.com/" />
+
+	<rule from="^http://(?:www\.)?allshareplay\.com/$"
```
Comment (J0WI) [inline src/chrome/content/rules/AllShare_Play.xml]: This is only needed for ^allshareplay.com/$, www works server side https. Please also document why this is needed (403).

File: src/chrome/content/rules/Samsung.com.xml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+			<test url="http://www.samsung.com/us/support/" />
+
+	<!--	Not secured by server:
+					-->
+	<!--securecookie host="^\.samsung\.com$" name="^(_common_country|_common_lang|COUNTRY_CODE|EXPAND_SCHEDULED|REMEMBERED_DEVICE_CODE_CAN)$" /-->
+	<!--securecookie host="^apps\.samsung\.com$" name="^WMONID$" /-->
+
+	<securecookie host="^(?!shop\.us\.).*\.samsung\.com$" name=".+" />
+
+	<target host="*.samsung.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung.com.xml]: It's a bit irritating to have a wildcard target and that much exclusions. I would prefer listing the targets, the you can also get rid of the 'Fully covered domains' list on the top.

File: src/chrome/content/rules/Samsung.com.xml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+	<rule from="^http://www\.samsung\.com/"
+		to="https://www.samsung.com/" />
+		<test url="http://www.samsung.com/cn/common/css/home.css" />
+		
+	<rule from="^http://nmetrics\.samsung\.com/"
+		to="https://samsung-com.112.2o7.net/" />
+		<test url="http://nmetrics.samsung.com/b/ss/sssamsung4in" />
+		
+	<rule from="^http://((?:stg-)?account|cdn|(?:help\.|img\.|pub\.)?content|findmymobile|images|link|secureus|smetrics|support-cn|us|shop\.us)\.samsung\.com/"
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung.com.xml]: quoting from our style guide:
> Prefer capturing groups (www\.)? over non-capturing (?:www\.)?. The non-capturing form adds extra line noise that makes rules harder to read. Generally you can achieve the same effect by choosing a correspondingly higher index for your replacement group to account for the groups you don't care about.

File: src/chrome/content/rules/Samsung.com.xml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+		<test url="http://cdn.samsung.com/common/next/css/apps/mobile/barshade.css" />
+		<test url="http://help.content.samsung.com/csweb/main/main.do" />
+		<test url="http://findmymobile.samsung.com/img/favicon02.ico/" />
+		<test url="http://images.samsung.com/" />
+		<test url="http://link.samsung.com/" />
+		<test url="http://secureus.samsung.com/us/mobile/galaxy-tab/SM-T810NZWEXAR" />
+		<test url="http://smetrics.samsung.com/" />
+		<test url="http://support-cn.samsung.com/cn/common/css/local_global_common.css" />
+		<test url="http://us.samsung.com/us/index.jsp" />
+		<test url="http://shop.us.samsung.com/store?Action=DisplayDrcartSummary" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung.com.xml]: Please also add test urls for:
stg-account.samsung.com
img.content.samsung.com
pub.content.samsung.com

File: src/chrome/content/rules/Samsung_Apps.xml [hunk 5]
```diff
@@ -42,33 +27,40 @@
...
-	<rule from="^http://(?:www\.)?samsungapps\.com/"
-		to="https://www.samsungapps.com/" />
-
-	<rule from="^http://(\w\w-odc|mkt-odc|tv)\.samsungapps\.com/"
+	<rule from="^http://(\w\w-odc|mkt-odc)\.samsungapps\.com/"
 		to="https://$1.samsungapps.com/" />
+		<test url="http://us-odc.samsungapps.com/" />
+		<test url="http://mkt-odc.samsungapps.com/" />
+
+	<rule from="^http://img\.samsungapps\.com/(.+)\.(apk|gif|pdf|swf|xls|xml)"
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Apps.xml]: you should have a `$` on the end of the regex.

File: src/chrome/content/rules/Samsung_Apps.xml [hunk 6]
```diff
@@ -42,33 +27,40 @@
...
+		<test url="http://img.samsungapps.com/support/Application_protection_guide.pdf" />
+		<test url="http://img.samsungapps.com/display/2012/0327/0610/uploadfile_20120327061003463.swf" />
+		<test url="http://img.samsungapps.com/news/1st_winnerlist.xls" />
+		<test url="http://img.samsungapps.com/common/countryList.xml" />
+		
+	<exclusion pattern="^http://img\.samsungapps\.com/$" />
+		<test url="http://img.samsungapps.com/" />
+
+	<rule from="^http://seller\.samsungapps\.com/"
+		to="https://seller.samsungapps.com/" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Apps.xml]: I get an error on seller.samsungapps.com/tv/* over https

File: src/chrome/content/rules/Samsung_Apps.xml [hunk 7]
```diff
@@ -42,33 +27,40 @@
 			- us-odc
 
 		- mkt-odc
+	
+	502:
 		- tv
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Apps.xml]: Mismatches on m.\w+.samsungapps.com

File: src/chrome/content/rules/Samsung_Fire.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+	Mismatch:
+	*.samsungfire.com
+-->
+
+<ruleset name="SamsungFire.com">
+
+	<target host="samsungfire.com" />
+	<target host="direct.samsungfire.com" />
+	<target host="www.samsungfire.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Fire.xml]: m.samsungfire.com

File: src/chrome/content/rules/Samsung_Fire.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<!--
+	For other Samsung coverage, see Samsung.com.xml.
+
+	Mismatch:
+	*.samsungfire.com
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Fire.xml]: These are (unsorted):

s.samsungfire.com
brand.samsungfire.com
sts.samsungfire.com
rm.samsungfire.com
direct.samsungfire.com
blog.samsungfire.com
gooddaddy.samsungfire.com
v3.samsungfire.com
us.samsungfire.com
smarket.samsungfire.com
volleyball.samsungfire.com
yuseong.samsungfire.com
weather.samsungfire.com
us.samsungfire.com
globalportal.samsungfire.com
find.samsungfire.com
etps.samsungfire.com
dmptprod.samsungfire.com
u.samsungfire.com
mdirect.samsungfire.com

File: src/chrome/content/rules/Samsung_Knox.xml [hunk 10]
```diff
@@ -0,0 +1,21 @@
+<!--
+	For other Samsung coverage, see Samsung.com.xml.
+
+	^: cert only matches *.samsungknox.com
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Knox.xml]: works for me and is included as target

File: src/chrome/content/rules/Samsung_Knox.xml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+	<target host="samsungknox.com" />
+	<target host="cdn.samsungknox.com" />
+	<target host="emm.samsungknox.com" />
+	<target host="emm1.samsungknox.com" />
+	<target host="enroll.samsungknox.com" />
+	<target host="me.samsungknox.com" />
+	<target host="my.samsungknox.com" />
+	<target host="support.samsungknox.com" />
+	<target host="www.samsungknox.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Knox.xml]: www2.samsungknox.com
custom.samsungknox.com

File: src/chrome/content/rules/Samsung.com.xml [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+		
+</ruleset>
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung.com.xml]: displaysolutions.samsung.com also works for me

File: src/chrome/content/rules/Samsung_Sem.xml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+	weblib.samsungsem.com
+
+-->
+
+<ruleset name="Samsung_Sem (partial)">
+
+	<target host="samsungsem.com" />
+	<target host="www.samsungsem.com" />
+	
+	<rule from="^http://(www\.)?samsungsem\.com/" to="https://www.samsungsem.com/" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Sem.xml]: please document the issue

File: src/chrome/content/rules/Samsung_Sem.xml [hunk 14]
```diff
@@ -0,0 +1,16 @@
+<!--
+	For other Samsung coverage, see Samsung.com.xml.
+
+Mismatch:
+	weblib.samsungsem.com
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Sem.xml]: same on thesis.

File: src/chrome/content/rules/Samsung_Techwin.xml [hunk 15]
```diff
@@ -1,25 +1,16 @@
...
 	<target host="samsungtechwin.co.kr" />
 	<target host="www.samsungtechwin.co.kr" />
 
-
-	<securecookie host="^www\.samsungtechwin\.co\.kr$" name=".+" />
-
-
 	<rule from="^http://(?:www\.)?samsungtechwin\.co\.kr/"
-		to="https://www.samsungtechwin.co.kr/" />
+		to="https://www.hanwhatechwin.co.kr/" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: Is this still part of Samsung?

File: src/chrome/content/rules/Samsung_Techwin.xml [hunk 16]
```diff
@@ -1,25 +1,16 @@
...
+<ruleset name="Samsung_Techwin">
 
 	<target host="samsungtechwin.co.kr" />
 	<target host="www.samsungtechwin.co.kr" />
 
-
-	<securecookie host="^www\.samsungtechwin\.co\.kr$" name=".+" />
-
-
 	<rule from="^http://(?:www\.)?samsungtechwin\.co\.kr/"
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: please avoid non-capturing (`?:`)

File: src/chrome/content/rules/Samsung_Techwin.xml [hunk 17]
```diff
@@ -1,25 +1,16 @@
...
+	For other Samsung coverage, see Samsung.com.xml.
+	
 	Nonfunctional domains:
-
-		- (www.)samsungtechwin.com	(shows co.kr; mismatched, CN: www.samsungtechwin.co.kr)
-
-
-	Cert only matches www.
-
+		- (www.)samsungtechwin.com	(to http://www.hanwhatechwin.com/)
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: Mismatches on blog. and eco.

File: src/chrome/content/rules/Samsung_Techwin.xml (hunk 15 above)
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: You may also include hanwhatechwin.co.kr as targets if it's the same.

File: src/chrome/content/rules/Samsung_mySingle.xml [hunk 18]
```diff
@@ -1,19 +1,34 @@
...
+<ruleset name="Samsung_mySingle (partial)">
 
 	<target host="samsung.net" />
+	<target host="light.samsung.net" />
+	<target host="m.samsung.net" />
+	<target host="scsi.sec.samsung.net" />
+	<target host="tcs.sec.samsung.net" />
+	<target host="sfts.samsung.net" />
+	<target host="ssdrm.samsung.net" />
 	<target host="www.samsung.net" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_mySingle.xml]: I get 401 response on ^ and www

**Comment by J0WI** [comment]
I finally found some time to have a look here. It's generally difficult to review such huge PRs that affect multiple files.

File: src/chrome/content/rules/Samsung_Techwin.xml (hunk 15 above)
Comment (renyouguo) [inline src/chrome/content/rules/Samsung_Techwin.xml]: Should I rename it to hanwhatechwin ?

**Comment by renyouguo** [comment]
@J0WI Thank you very much.

File: src/chrome/content/rules/AllShare_Play.xml [hunk 19]
```diff
@@ -1,33 +1,27 @@
...
+	<target host="allshareplay.com" />
+	<target host="www.allshareplay.com" />
+	<target host="fwk.allshareplay.com" />
+	<target host="m.allshareplay.com" />
+		<test url="http://m.allshareplay.com/pc/main/photoAlbumCreate.html" />
+	<!--	403 in http://m.allshareplay.com/$	-->
+	<exclusion pattern="^http://m\.allshareplay\.com/$" />
+		<test url="http://m.allshareplay.com/" />
+
+	<rule from="^http://allshareplay\.com/$"
```
Comment (J0WI) [inline src/chrome/content/rules/AllShare_Play.xml]: Please also document the 403 here.

File: src/chrome/content/rules/Samsung.com.xml [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+	Fully covered domains:
+
+		- samsung.com subdomains:
+
+			- (www.)	(^ → www)
+			- account
+			- allshareplay
+			- content
+			- img.content
+			- pub.content
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung.com.xml]: Why has img.content and pub.content has been removed from the rules?
If they have issues or do no loner exist, the comments should also be updated.

File: src/chrome/content/rules/Samsung.com.xml [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+
+	<target host="www.samsung.com" />
+
+	<exclusion pattern="^http://www\.samsung\.com/$" />
+
+	<!-- Support page FAQ and other features are broken. Also, their i18n is a lie.
+		Github issue: https://github.com/EFForg/https-everywhere/issues/309 -->
+	<!--	MCB : https://www.samsung.com/cn|tw|us|.*/apps/mobile/	-->	
+
+	<rule from="^http://www\.samsung\.com/(.+)\.(css|gif|ico|js|png|svg|woff)"
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung.com.xml]: You may also add a `$` on the end of the regex here.

File: src/chrome/content/rules/Samsung_Fire.xml [hunk 22]
```diff
@@ -1,15 +1,34 @@
...
+	<target host="smarket.samsungfire.com" />
+	<target host="sts.samsungfire.com" />
+	<target host="u.samsungfire.com" />
+	<target host="us.samsungfire.com" />
+	<target host="us.samsungfire.com" />
+	<target host="v3.samsungfire.com" />
+	<target host="volleyball.samsungfire.com" />
+	<target host="weather.samsungfire.com" />
 	<target host="www.samsungfire.com" />
+	<target host="yuseong.samsungfire.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Fire.xml]: please double check the added hosts, I think ^, direct., m. and www. were the only one that works.

File: src/chrome/content/rules/Samsung_Techwin.xml [hunk 23]
```diff
@@ -1,25 +1,25 @@
...
 	<target host="www.samsungtechwin.co.kr" />
-
-
-	<securecookie host="^www\.samsungtechwin\.co\.kr$" name=".+" />
-
-
-	<rule from="^http://(?:www\.)?samsungtechwin\.co\.kr/"
-		to="https://www.samsungtechwin.co.kr/" />
-
+	<!--	Redirect to new host by serve.	-->
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: Typo: serve => server

File: src/chrome/content/rules/Samsung_Techwin.xml [hunk 24]
```diff
@@ -1,25 +1,25 @@
...
+	<rule from="^http://(www\.)?samsungtechwin\.co\.kr/"
+		to="https://www.hanwhatechwin.co.kr/" />
+
+	<target host="hanwhatechwin.co.kr" />
+	<target host="www.hanwhatechwin.co.kr" />
+	<rule from="^http://(www\.)?hanwhatechwin\.co\.kr/"
```
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: Could you document why the redirect from ^ to www is needed?

File: src/chrome/content/rules/Samsung_Techwin.xml (hunk 15 above)
Comment (J0WI) [inline src/chrome/content/rules/Samsung_Techwin.xml]: Yes, please. Do not forget to update the reference in Samsung.com.xml

**Comment by J0WI** [comment]
Just a few minor issues left, see inline comments.

**Comment by J0WI** [comment]
LGTM, thanks for this huge contribution!

**Pull request merged**